File: problems/Reverse String/reverseString.cs
public class Solution {
    public void ReverseString(char[] s) {
        var n = s.Length - 1;
        var half = s.Length >> 1;
        var idx = 0;

        while (half > idx) {
            var letter = s[idx];
            s[idx] = s[n - idx];
            s[n - idx] = letter;
            ++idx;
        }
    }
}

File: problems/Reverse String/reverseString.java
class Solution {
    public void reverseString(char[] s) {
        int n = s.length;

        for (int idx = 0; n >> 1 > idx; ++idx) {
            char tmp = s[idx];
            s[idx] = s[n - idx - 1];
            s[n - idx - 1] = tmp;
        }
    }
}
